### ISSUE TITLE
Prevent print-only components from rendering when we aren't in the 'print mode'.

### ISSUE THREAD

**Issue by alex-gokkun** [a-x-/react-easy-print#39]
Correct me if I'm wrong but I haven't found an official way for a print-only component not being rendered. 
Let's say that I have a component that depends on a plethora of props. Each prop change would trigger a re-render. It's a waste of effort if I'm not actually printing anything at the moment. 
By subscribing to _beforeprint_ and _afterprint_ events I got a workaround but I cannot say I'm satisfied with it. It seems to violate separation of conerns.
Are there any plans on supporting this in the library?

**Comment by a-x-** [comment]
Hm, official way is there. Let me a second

**Comment by a-x-** [comment]
Did you try `<Print printOnly...`?

**Comment by alex-gokkun** [comment]
Yes, I did. It prevents the component from being displayed but doesn't prevent it from being rendered.

**Comment by a-x-** [comment]
I got now. Thank you for the case!

There is reason for this behavior. 
If PrintProvider would re-render the children (whole components tree)
state of components will be lost. 

I thinked a few minutes right now and realized the solution: 
Through pubsub in the context 
trigger target re-renderings for printOnly and some another components
(Like redux does it in its connect hocs).

Do you want try to implement this?

I can try in a 2-4 next days.

**Comment by AustinMutschler** [comment]
Any update on this?
